### ISSUE TITLE
Internal error when indexing

### ISSUE THREAD

**Issue by bb** [meilisearch/MeiliSearch#1788]
**Describe the bug**

TLDR: I found a trivial json with two docs which triggers a panic, see section "To Reproduce".

I got this error in MeiliSearch 0.23.0rc2:

```
meilisearch_1  | 2021-10-08T15:09:25.177958000Z thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Custom { kind: Other, error: "keys must be inserted in order and only one time" })', /meilisearch/meilisearch-lib/src/document_formats.rs:90:38
```

The ruby client reports this:

```
MeiliSearch::ApiError: 500 Internal Server Error - Internal error: panic. See https://docs.meilisearch.com/errors#internal.
from /Users/bb/.rvm/gems/ruby-3.0.2/gems/meilisearch-0.16.1/lib/meilisearch/http_request.rb:79:in `validate'
```

What I was doing:
```ruby
    index.add_documents(documents)
```
Nothing showed up in the updates queue.

With meilisearch 0.22.0 this was still working, resulting in an update entry like this:
```
{"status"=>"processed", "updateId"=>23, "type"=>{"name"=>"DocumentsAddition", "number"=>5942}, "duration"=>1.686, "enqueuedAt"=>"2021-10-08T15:22:52.107452300Z", "processedAt"=>"2021-10-08T15:22:53.799640600Z"}
```
and a log entry like this
```
meilisearch_1  | 2021-10-08T15:22:53.485330100Z [2021-10-08T15:22:53Z INFO  meilisearch_http::index::updates] document addition done: DocumentAdditionResult { nb_documents: 5942 }
```

My first guess was, that it might be related to Geo stuff, because I already added fields like this months ago, in anticipation of the Geosearch feature…
```ruby
:_geoloc=>{:lat=>41.072348, :lng=>14.331134}
```

However, I just checked https://github.com/meilisearch/specifications/blob/develop/text/0059-geo-search.md and see the field name should be _geo so it might be something else. I also made sure it's not the `geoloc` field by renaming it to `x_geoloc`, just in case _geoloc might also be supported.

**To Reproduce**

```bash
curl -i -X POST 'http://localhost:7700/indexes/foo/documents' \
     -H 'Content-Type: application/json' \
     --data '[{"id": 1,"b": 0},{"id": 2,"a": 0,"b": 0}]'
```

**Expected behavior**

I expect successful indexing or at least a helpful error message, ideally in the updates queue and not only in the server logs.

**MeiliSearch version:** [e.g. v0.23.0rc2]

**Additional context**

~~I didn't yet try to extract test data to reproduce, but happy to provide some if necessary. Hit me up here or in Slack if you need something.~~ Edit: Added curl example above.

**Comment by MarinPostma** [comment]
@curquiza we may want to issue a fix on Monday before releasing?
